src/main/java/teammates/ui/controller/StudentProfileEditSaveAction.java

```diff
@@ -52,7 +53,8 @@ private StudentProfileAttributes extractProfileData() {
         if ("".equals(editedProfile.nationality)) {
             editedProfile.nationality = getRequestParamValue("existingNationality");
         }
-        editedProfile.gender = getRequestParamValue(Const.ParamsNames.STUDENT_GENDER);
+
+        editedProfile.gender = Gender.valueOf(getRequestParamValue(Const.ParamsNames.STUDENT_GENDER).toUpperCase());
```

Comment:
While unlikely to happen, an invalid value here is going to cause a runtime exception. This should be handled.

Reply:
sir in Last Review You have told me that From UI we will force the user To Select only Gender As MALE ,FEMALE ,OTHER so there would not be any case of Invalid Gender Type And Also I had made Changes in test cases to eliminate this problem So invalid value of gender would never occur In normal circumtances

Reply:
Does not occur _under normal circumstances_ does not mean it will never occur.

Reply:
As gender is taken input from the radio Button so There is no chance of illegal gender type . But no problem sir ( @wkurniawan07 ) I will fix this

Reply:
It's simple for a tech-savvy user to tinker with the html of the page to send a different value. Back-end code should not assume the front end is working perfectly.